src/sentry/lang/native/symbolicator.py

```diff
@@ -250,27 +259,31 @@ def parse_sources(config):
     return sources
 
 
+def get_options_for_project(project):
+    assert project._organization_cache is not None
+    return {
+        # Symbolicators who do not support options will ignore this field entirely.
+        "dif_candidates": features.has("organizations:dif-candidates", project.organization)
+    }
+
+
 def get_sources_for_project(project):
     """
     Returns a list of symbol sources for this project.
     """
 
     sources = []
 
-    if not getattr(project, "_organization_cache", False):
-        with sentry_sdk.start_span(op="lang.native.symbolicator.organization.get_from_cache"):
-            project._organization_cache = Organization.objects.get_from_cache(
-                id=project.organization_id
-            )
-
     # The symbolicator evaluates sources in the order they are declared. Always
     # try to download symbols from Sentry first.
     project_source = get_internal_source(project)
     sources.append(project_source)
 
     # Check that the organization still has access to symbol sources. This
     # controls both builtin and external sources.
+    assert project._organization_cache is not None
```

Comment:
I would prefer to remove such assertions. While they are helpful in development, there is no functional issue or violated invariant here. If however, the assertion is violated, we crash processing, which must not happen.

Reply:
it's almost impossible to hit this assertion in production and not in tests, but it's possible to introduce performance regressions that are only caught in prod

Reply:
Introducing a possible crash into production for the chance of catching a potential performance regression is not worth the risk IMHO. We have ways of monitoring performance without crashing.